mubench.pipeline/tests/data/test_detector_run.py

```diff
@@ -183,3 +183,13 @@ def test_run_outdated_on_newer_compile(self, read_yamls_mock, _):
         uut = DetectorRun(self.detector, self.version, self.findings_path)
 
         assert uut._newer_compile(compile_timestamp)
+
+    @patch("data.detector_run.DetectorRun._load_findings")
+    @patch("data.detector_run.open_yamls_if_exists")
+    def test_get_run_info_without_previous_load(self, read_yamls_mock, load_findings_mock, _):
+        read_yamls_mock.return_value = {}
+        load_findings_mock.return_value = []
+
+        uut = DetectorRun(self.detector, self.version, self.findings_path)
+
+        run_info = uut.get_run_info()
```

Comment:
I'm not sure what the purpose of this test is. There's no assertion?

Reply:
There's a `dict.update(None)` happening, and I wasn't sure if that even works. In actual runs, the internal `self.__run_info` would be loaded at that point, but that's not guaranteed in any way. This test just makes sure it doesn't crash, if we request the run_info before it was loaded internally. It's a bit hard to explain and may be unnecessary. That's why I made a separate commit for this change; we could just drop it.

Reply:
I've added an assert which makes the test a bit more useful. It's essentially checking the default now.

Reply:
I'm not sure I grasp the `None` scenario. Where is the `None` coming from? There shouldn't be `None`s flying around anywhere, I think. Is it the test setup that triggers this? But the line 190 should ensure that loading the run info returns an empty dictionary. Can you say a bit more? (Or we can also discuss this on Thursday. It's not critical, I just want to understand).

Reply:
When a `DetectorRun` is initialized, we set `self.__run_info = None`. 
This value is only set in two places:
* `__get_run_info(...)`: Load the run info and then acts like a `dict.get(key, default)` and is only called on any direct access on values, e.g. `run.result`.
* `__save_run_info(...)`: Updates run info after write. 

In practice, we always call these at least once before requesting the full run info during publish. The test case was just there to make sure that, if there was any path where none of these were called before, we don't crash. The reason I wasn't sure about that, is this function:

```
    def get_run_info(self):
        run_info = {
            "number_of_findings": self.number_of_findings,
            "runtime": self.runtime,
        }
        run_info.update(self.__run_info)
        return run_info
```

In the scenario of the test case, `self.__run_info` is only accidentally not `None`, because it is loaded on the `self.runtime` call. The test was just there to make sure that this won't break if someone changes the code, as that seems very implicit to me. For example, with the following code, the test will fail:

```
    def get_run_info(self):
        run_info = {
            "number_of_findings": self.number_of_findings,
        }
        run_info.update(self.__run_info)
        return run_info
```

We could probably change `self.__run_info` to a property, which ensures it was loaded before access. This would also make the `self.__get_run_info` obsolete, as we could use `self.__run_info.get` directly. 

Anyway, I thought since it works, a test to make sure it stays working is enough.

Reply:
Thanks for the clarification. Now I understand.

Since, apparently, I wasn't aware of this potentially problematic behaviour when I implemented `get_run_info()`, I would prefer to fix this before it actually causes problems. I think using a property like you suggest is the right way to do this. Could you take care of this? You can add it to this PR and I will separate out the two changes.